### ISSUE TITLE
USWDS - Bug: Improve keyboard interactions in date picker

### ISSUE THREAD

**Issue by amyleadem** [uswds/uswds#5505]
## Summary
This issue originates from [this comment](https://github.com/uswds/uswds/pull/5374#pullrequestreview-1615663080) in https://github.com/uswds/uswds/pull/5374:

The date picker component has some quirks with keyboard navigation in VoiceOver:

- :warning: When VoiceOver is turned on in Chrome and Safari, the up and down arrows stop working as expected after using the `Tab` key anywhere in the component. (This matches the behavior in `develop` and also occurs in your [div-based demo](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-date-picker-remove-table/iframe.html?args=&id=components-form-inputs-date-picker--default&viewMode=story))
    - Note: This issue does NOT happen when using `VO` + `Up/down arrow` (This is an improvement over `develop`)
- :warning: When VoiceOver is turned on in Firefox, the up and down arrows close the dialog when VoiceOver is turned on (This matches the behavior in `develop` and also occurs in your [div-based demo](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-date-picker-remove-table/iframe.html?args=&id=components-form-inputs-date-picker--default&viewMode=story))
    - Note: This issue does NOT happen when using `VO` + `Up/down arrow` (This is an improvement over `develop`)
- :warning: When VoiceOver is turned on in Firefox and VoiceOver Gestures is turned on in iOS Safari, you cannot move to another month with the left/right arrows. You must use the top navigation to move to next/previous month. (This matches the behavior in `develop`)
